identity/src/commonMain/kotlin/com/android/identity/util/ByteArrayUtil.kt

```diff
@@ -0,0 +1,126 @@
+package com.android.identity.util
+
+/**
+ * ByteArray extension functions for consistent reading and writing commonly used types.
+ */
+fun ByteArray.putInt8(value: Int, offset: Int = 0) {
+    require(value in Byte.MIN_VALUE..Byte.MAX_VALUE) { "Value $value is out of Int8 range" }
+    this[offset] = value.toByte()
+}
+
+fun ByteArray.putInt16(value: Int, offset: Int = 0) {
+    require(value in Short.MIN_VALUE..Short.MAX_VALUE) { "Value $value is out of Int16 range" }
+    this[offset] = (value and 0xFF).toByte() // Lower byte.
+    this[offset + 1] = ((value shr 8) and 0xFF).toByte()  // Higher byte.
+}
+
+fun ByteArray.putInt32(value: Int, offset: Int = 0) {
+    this[offset] = (value and 0xFF).toByte()
+    this[offset + 1] = ((value shr 8) and 0xFF).toByte()
+    this[offset + 2] = ((value shr 16) and 0xFF).toByte()
+    this[offset + 3] = ((value shr 24) and 0xFF).toByte()
+}
```

Comment:
Your implementations are all for Little-Endian so it won't work out of the box for e.g. Cbor.kt which is Big Endian / Network Byte Order. So we definitely need to support both and our preference should be for Big Endian which is more commonly used.

I would suggest having both getXXX() / getXXXLe() and putXXX() / putXXXLe() variants.